server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3HealthCheck.java

```diff
@@ -0,0 +1,61 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.objectstorage.aws;
+
+import java.time.Duration;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.apache.james.blob.api.BlobStoreDAO;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.core.healthcheck.ComponentName;
+import org.apache.james.core.healthcheck.HealthCheck;
+import org.apache.james.core.healthcheck.Result;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class S3HealthCheck implements HealthCheck {
+
+    private static final Integer HEALTH_CHECK_TIMEOUT = 5;
+
+    private static final ComponentName COMPONENT_NAME = new ComponentName("S3");
```

Comment:
`ObjectStore` is a better wording IMO

Reply:
Currently I put object storage health check file in blob-s3 module. However, because this is generic health check for storage object, not s3 alone, I think I should move to it to either blob-api or blob-common. Where do you think it should be?

Reply:
I moved object storage health check file to blob-s3 module